config-object-store/src/main/java/org/hypertrace/config/objectstore/IdentifiedFilterPushedDownObjectStore.java

```diff
@@ -72,19 +74,29 @@ public Optional<T> getMatchingData(RequestContext context, F filterInput, List<S
     return getMatchingObject(context, filterInput, sortInput).map(ConfigObject::getData);
   }
 
-  public ConfigsResponse<T> getMatchingObjectsWithTotalCount(
-      RequestContext context, F filterInput, List<S> sortInput, Pagination pagination) {
+  public ConfigsResponse<ContextualConfigObject<T>> getMatchingObjectsWithTotalCount(
+      RequestContext context, F filterInput, List<S> sortInput, @Nullable Pagination pagination) {
+    Filter filter = buildFilter(filterInput);
```

Comment:
The filter and sort conversion can move inside the shared method too I believe

Reply:
fixed it